docs/devGuide/migratingToTypeScript.md

```diff
@@ -0,0 +1,135 @@
+
+   However, in some cases, changing to the ES6 syntax might be necessary. For instance, you cannot use the TypeScript equivalent syntax if you want to import/export TypeScript specific constructs, such as types, interfaces, and enums. Again, when doing so, make sure that all tests pass.
+
+   <box type="info">
+
+   We will perform a mass change into ES6 syntax after all files in core have been adapted into TypeScript.
+   </box>
+
+3. **Adapt the files** to fully use TypeScript.
+
+   The errors from TypeScript and `typescript-eslint` can guide you on what to fix. Only fix what is necessary and do not modify any code functionality. Avoid using `any` as best as you can.
+
+   If you happen to encounter type errors related to using names from an internal dependency that is still in JavaScript (`.js`), you can try to infer simple types in your file first as a "stand-in" of the more robust types when that internal dependency is eventually migrated.
+
+   On the flip side of the above situation, as you now have developed a robust type for your own file, you might have to **adjust how other TypeScript (`.ts`) files type names referenced from your files**. Some type declarations might need to be made stricter, or can be removed if it's not necessary anymore, and so on. This will ensure that the files will be incrementally stricter. 
+
+4. **Make sure everything is in order** by running `npm run test` in the root directory.
+
+   The `core` package test directly tests `.ts` files (powered by `ts-jest`), and the `cli` package test uses the compiled files from `core`. Therefore, passing the `cli` tests means that it is very likely the compiled files work as expected.
+
+5. **Stage the adapted files and the typings from Step 1** for commit. Verify the changes before committing.
+
+6. **Commit the changes** with the message `Adapt <file-or-folder> to TypeScript`.
+   <br>
+   For example, `Adapt core/src/html to TypeScript`.
+
+You are now ready to create a pull request for the changes to the repository.
+
+## Example of Migrated Works
+
+You can see these pull requests to observe the finished migration works:
+
+- [#1877: Adopt TypeScript for core package](https://github.com/MarkBind/markbind/pull/1877)
+
+## Explanation Notes
+
+### Why do we need separate "Rename" and "Adapt" commits
+
+We want to keep the file history intact throughout this migration, meaning that we should still be able to see the change history of the `.ts` files even from the time when they were still in `.js`.
+
+When we rename files, we have to keep the Git similarity index between the old file and the new file to be above a certain threshold (50%) to keep the history, or else it will be regarded as a different file with its own history.
```

Comment:
```suggestion
When we rename files, we have to keep the Git similarity index between the old file and the new file above a certain threshold (50%) to keep the history, or else it will be regarded as a different file with its own history.
```